### PR TITLE
fix: parsing numbers more than i32 and u32

### LINKED ISSUE #85: Error: out of range integral type conversion attempted
When I try deserialize next value, I've got an error "out of range integral type conversion attempted".
```rs
    let json = r#"{"num": 5000000000}"#;
    let obj: serde_json::Value = serde_json::from_str(json)?;
    println!("num {}", obj["num"]);

    let vec = serde_json::to_vec(&obj).unwrap();
    let pb = serde_json::from_slice::<pbjson_types::Value>(&vec).unwrap();
    println!("{:#?}", pb);
```
This happens because next [function](https://github.com/influxdata/pbjson/blob/main/pbjson-types/src/value.rs#L186-L191) throw an error:
```rs
    fn visit_u64<E>(self, v: u64) -> Result<Self::Value, E>
    where
        E: de::Error,
    {
        self.visit_u32(v.try_into().map_err(de::Error::custom)?)
    }
```
If look [visit_u32](https://github.com/influxdata/pbjson/blob/7197cb4d44643be0bcf94cb925ceb57223c78edf/pbjson-types/src/value.rs#L179-L184) function, so number value wrapped in Kind::NumberValue which accept f64 type:
```rs
    fn visit_u32<E>(self, v: u32) -> Result<Self::Value, E>
    where
        E: de::Error,
    {
        Ok(Kind::NumberValue(v.into())) // f64
    }
```
As a result, the current conversion pipeline looks like u64 -> u32 -> f64. If it were u64 -> f64 (without u32), then we would not lose a significant range of numerical values (u32, f64].

For example like this:
```rs
    fn visit_u64<E>(self, v: u64) -> Result<Self::Value, E>
    where
        E: de::Error,
    {
        Ok(Kind::NumberValue(
            v.to_string().parse::<f64>().map_err(de::Error::custom)?,
        ))
    }
```

### PR DESCRIPTION
**Pull request by NAlexandrov**
fix #85

### CODE REVIEW

**Comment by dmah42** [comment]
is there anything i can do to help this PR along?

File: pbjson-types/src/value.rs [hunk 1]
```diff
@@ -187,14 +193,20 @@ impl<'de> serde::de::Visitor<'de> for KindVisitor {
     where
         E: de::Error,
     {
-        self.visit_u32(v.try_into().map_err(de::Error::custom)?)
+        if v < 1 << 53 {
```
Comment (tustvold) [inline pbjson-types/src/value.rs:196]:
```suggestion
        if v < 1 << f64::MANTISSA_DIGITS {
```

**Review by tustvold — APPROVED** [review]
Apologies this slipped off my radar somehow, just some minor nits.

Could I also ask you to please sign the [CLA](https://www.influxdata.com/legal/cla/) and I can then get this in.

Edit: looking into CI failures (CI should run and work if you push another commit)

File: pbjson-types/src/value.rs [hunk 2]
```diff
@@ -152,14 +152,20 @@ impl<'de> serde::de::Visitor<'de> for KindVisitor {
     where
         E: de::Error,
     {
-        self.visit_i32(v.try_into().map_err(de::Error::custom)?)
+        if v > -(1 << 53) && v < 1 << 53 {
```
[the remark this answers is not in the document]
Comment (NAlexandrov) [reply pbjson-types/src/value.rs]: fixed

File: pbjson-types/src/value.rs (hunk 1 above)
Comment (NAlexandrov) [reply pbjson-types/src/value.rs]: fixed

**Comment by NAlexandrov** [comment]
> Could I also ask you to please sign the [CLA](https://www.influxdata.com/legal/cla/) and I can then get this in.

done

**Comment by tustvold** [comment]
Sorry this took quite so long to get to... I've been a bit distracted with other projects :sweat_smile:

**Pull request merged**